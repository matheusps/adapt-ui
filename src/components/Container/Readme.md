Simple example

```js
<Container>
  <p>This quark looks cool!</p>
</Container>
```
