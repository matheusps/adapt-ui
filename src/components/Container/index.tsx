import React, { FunctionComponent } from 'react'
import styled from 'styled-components'
import { selectTheme } from '../../global/helpers'

interface Props extends EnhancedWithTheme {}

const Div = styled.div<EnhancedWithTheme>`
  background-color: ${props =>
    selectTheme(props.theme, props.color, props.shade)};
  padding: 1rem;
`

const Container: FunctionComponent<Props> = ({ color, shade, children }) => {
  return (
    <Div color={color} shade={shade}>
      {children}
    </Div>
  )
}

Container.defaultProps = {
  color: 'base',
  shade: 'default',
}

export default Container
