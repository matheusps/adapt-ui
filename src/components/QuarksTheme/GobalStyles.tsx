import { createGlobalStyle } from 'styled-components'

/** HKGrotesk */
import FontLight from './font/light.woff'
import FontLight2 from './font/light.woff2'
import FontRegular from './font/regular.woff'
import FontRegular2 from './font/regular.woff2'
import FontMedium from './font/medium.woff'
import FontMedium2 from './font/medium.woff2'
import FontSemibold from './font/semiBold.woff'
import FontSemibold2 from './font/semiBold.woff2'
import FontBold from './font/bold.woff'
import FontBold2 from './font/bold.woff2'

const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: HkGrotesk;
    src: url(${FontLight}) format('woff'),
         url(${FontLight2}) format('woff2');
    font-weight: 300;
  }

  @font-face {
    font-family: HkGrotesk;
    src: url(${FontRegular}) format('woff'),
         url(${FontRegular2}) format('woff2');
    font-weight: 400;
  }

  @font-face {
    font-family: HkGrotesk;
    src: url(${FontMedium}) format('woff'),
         url(${FontMedium2}) format('woff2');
    font-weight: 500;
  }

  @font-face {
    font-family: HkGrotesk;
    src: url(${FontSemibold}) format('woff'),
         url(${FontSemibold2}) format('woff2');
    font-weight: 600;
  }

  @font-face {
    font-family: HkGrotesk;
    src: url(${FontBold}) format('woff'),
         url(${FontBold2}) format('woff2');
    font-weight: 800;
  }
  
  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  html {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  body {
    font-family: HkGrotesk, sans-serif;
    overflow-x: hidden;
    text-rendering: optimizeLegibility;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
    font-variant-ligatures: no-common-ligatures;
    padding: 0;
    margin: 0;
  }

  h1, h2, h3, h4, h5, h6, p, button, input { 
    font-family: HkGrotesk, sans-serif;
  }
`
export default GlobalStyles
