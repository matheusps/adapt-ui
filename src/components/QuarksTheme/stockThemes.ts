export const light = {
  colors: {
    base: {
      default: '#fafafa',
      darken: '#cecece',
      lighten: '#ffffff',
      contrast: '#000000',
    },
    primary: {
      default: '#191817', //EERIE BLACK
      darken: '#020100', //REGISTRATION BLACK
      lighten: '#474645', //OUTER SPACE
      contrast: '#F2EFE9', //ALABASTER
    },
  },
}
