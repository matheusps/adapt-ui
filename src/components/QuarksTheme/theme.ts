export const darkside: ThemeProps = {
  colors: {
    substratum: {
      r: 3,
      g: 3,
      b: 5,
    },
    contrast: {
      r: 242,
      g: 239,
      b: 233,
    },
    skin: {
      primary: {
        r: 100,
        g: 181,
        b: 246,
      },
      secondary: {
        r: 224,
        g: 224,
        b: 224,
      },
      success: {
        r: 129,
        g: 199,
        b: 132,
      },
      warning: {
        r: 255,
        g: 183,
        b: 77,
      },
      danger: {
        r: 240,
        g: 98,
        b: 146,
      },
    },
  },
  elements: {
    roundness: '.8rem',
  },
}
