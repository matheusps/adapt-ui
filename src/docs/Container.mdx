---
## name: Container
---

import Container from '../components/Container'

# Container

Represents a flexbox container

<Container>Some cool text</Container>
