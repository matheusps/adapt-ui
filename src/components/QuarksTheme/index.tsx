import React, { FunctionComponent } from 'react'
import { ThemeProvider } from 'styled-components'

import { light } from './stockThemes'

interface Props {
  theme: ThemeProps
}

/**
 * Wraps around Theme-provider from styled-components 💅🏻
 */
const QuarksTheme: FunctionComponent<Props> = ({ theme, children }) => {
  return (
    <ThemeProvider theme={theme}>
      <>{children}</>
    </ThemeProvider>
  )
}

QuarksTheme.defaultProps = {
  theme: light,
}

export default QuarksTheme
