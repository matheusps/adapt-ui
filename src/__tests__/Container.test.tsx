import React from 'react'
import { render } from 'react-testing-library'
import Container from '../components/Container'
import QuarksTheme from '../components/QuarksTheme'

describe('Container component', () => {
  const component = render(
    <QuarksTheme>
      <Container />
    </QuarksTheme>
  )

  it('should render', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
