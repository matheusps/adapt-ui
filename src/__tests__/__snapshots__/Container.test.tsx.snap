// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa iWgDvT"
  />
</DocumentFragment>
`;
